sympy/physics/mechanics/body.py

```diff
@@ -214,3 +214,106 @@ def apply_torque(self, vec):
         if not isinstance(vec, Vector):
             raise TypeError("A Vector must be supplied to add torque.")
         self.loads.append((self.frame, vec))
+
+    def masscenter_vel(self, body):
+        """
+        Returns the velocity of self's masscenter wrt body's referenceframe or
+        self's referenceframe.
+
+        Parameters
+        ==========
+
+        body: Body/ReferenceFrame
+            The body about which the velocity is needed to be calculated.
+            If body is self by default.
+
+        Example
+        =======
+
+        >>> from sympy.physics.mechanics import Body
+        >>> A = Body('A')
+        >>> B = Body('B')
+        >>> A.masscenter.set_vel(B.frame, 5*B.frame.x)
+        >>> A.masscenter_vel(B)
+        5*B_frame.x
+        >>> A.masscenter_vel(B.frame)
+        5*B_frame.x
+
+        """
+
+        if isinstance(body,  ReferenceFrame):
+            frame=body
+        elif isinstance(body, Body):
+            frame = body.frame
+        return self.masscenter.vel(frame)
+
+    def ang_vel_in(self, body):
+        """
+        Returns the angular velocity of self's frame wrt body's referenceframe or
+        self's referenceframe.
+
+        Parameters
+        ==========
+
+        body: Body
+            The body about which the angluar velocity is needed to be calculated.
+            If body is self by default.
+
+        Example
+        =======
+
+        >>> from sympy.physics.mechanics import Body, ReferenceFrame
+        >>> A = Body('A')
+        >>> N = ReferenceFrame('N')
+        >>> B = Body('B', frame=N)
+        >>> A.frame.set_ang_vel(N, 5*N.x)
+        >>> A.ang_vel_in(B)
+        5*N.x
+        >>> A.ang_vel_in(N)
+        5*N.x
+
+        """
+
+        if isinstance(body,  ReferenceFrame):
+            frame=body
+        elif isinstance(body, Body):
+            frame = body.frame
+        return self.frame.ang_vel_in(frame)
+
+    def dcm(self, body):
+        """
+        Returns the direction cosine matrix relative to the provided body's
+        reference frame.
```

Comment:
Returns the direction cosine matrix of this body relative to the provided rigid body or reference frame.